File: QuSoC/QuSoC/apps/MemBlock/source/MemBlock.cs
namespace MemBlock
{
    public class SOC
    {
        public static SOC Instance { get; set; } = new SOC();

        public virtual uint Counter { get; set; }
        public virtual uint[] MemBlock { get; set; } = new uint[1024];
    }

    public static class Firmware
    {
        public static void EntryPoint()
        {
            SOC.Instance.MemBlock[1023] = 42;

            for (byte i = 0; i < 10; i++)
            {
                SOC.Instance.MemBlock[i] = i;
            }

            for (byte i = 0; i < 10; i++)
            {
                SOC.Instance.MemBlock[1023] += SOC.Instance.MemBlock[i];
            }

        }
    }
}

File: QuSoC/QuSoC/apps/MemBlock/firmware/MemBlock.cpp
#include "MemBlock.h"
#include <stdint.h>
#include <stdbool.h>
#include "soc.h"
namespace MemBlock
{
	void Firmware::EntryPoint()
	{
		MemBlock_SOC_MemBlock[1023] = 42;
		for(uint8_t i = 0; (i < 10); (i++))
		{
			MemBlock_SOC_MemBlock[i] = i;
		}
		for(uint8_t i = 0; (i < 10); (i++))
		{
			MemBlock_SOC_MemBlock[1023] += MemBlock_SOC_MemBlock[i];
		}
	}
}
